tldw_Server_API/tests/Agent_Client_Protocol/test_mcp_transport.py

```
@@ -0,0 +1,102 @@
+pytestmark = pytest.mark.unit
+
+
+def test_mcp_transport_is_abstract():
+    """Instantiating MCPTransport directly should raise TypeError."""
+    with pytest.raises(TypeError):
+        MCPTransport()
+
+
+def test_create_transport_unknown_raises():
+    """create_transport should raise ValueError for unknown protocol."""
+    with pytest.raises(ValueError, match="Unknown.*protocol"):
```

Ruff (0.15.6)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)